carberretta/extensions/gateway.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2020-2021, Carberra Tutorials
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import datetime as dt
+import typing as t
+
+import hikari
+from lightbulb import plugins
+
+if t.TYPE_CHECKING:
+    from lightbulb.app import BotApp
+
+from carberretta import Config
+
+TIMEOUT = 600
+
+plugin = plugins.Plugin("Gateway")
+
+
+async def schedule_action(member: hikari.Member, secs: int = TIMEOUT) -> None:
+    async def _take_action(member: hikari.Member) -> None:
+        if member.is_pending:
+            return await member.kick(
+                reason=(
+                    "Member failed to accept the server rules before "
+                    "being timed out."
+                )
+            )
+
+        for role in [Config.ANNOUNCEMENTS_ROLE_ID, Config.VIDEOS_ROLE_ID]:
+            await plugin.bot.rest.add_role_to_member(
+                Config.GUILD_ID,
+                member,
+                role,
+                reason="Member accepted the server rules.",
+            )
+
+    plugin.bot.d.scheduler.add_job(
+        _take_action,
+        id=f"{member.id}",
+        next_run_time=dt.datetime.utcnow() + dt.timedelta(seconds=secs),
+        args=[member],
+    )
+
+
+@plugin.listener(hikari.StartedEvent)
+async def on_started(_: hikari.StartedEvent) -> None:
+    async for m in plugin.bot.rest.fetch_members(Config.GUILD_ID):
+        if (secs := (dt.datetime.utcnow() - m.joined_at).seconds) <= TIMEOUT:
```

Comment:
```suggestion
    now = dt.datetime.now().astimezone()
    
    async for m in plugin.bot.rest.fetch_members(Config.GUILD_ID):
        if (secs := (now - m.joined_at).seconds) <= TIMEOUT:
```
This needs to be changed as Hikari uses offset-aware datetime objects, unlike discord.py.

Reply:
Fixed as well, I wasn't aware Hikari did that